src/arrayMethodJoin.js

```diff
@@ -4,8 +4,33 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinResultString = '';
+
+    if (this.length < 0) {
+      return joinResultString;
+    }
```

Comment:
```suggestion
    if (this.length < 0) {
      return joinResultString;
    }
```
It's redundant. You are returning `joinResultString` either way. If this has no length, the loop will not start and you eventually return the empty string